### PR TITLE
MDCT-2715: Initial Zustand Setup

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Setting up `Zustand` as our state management library for MFP. exchanging the `useUser` context hook with `useStore`.

### Related ticket(s)

MDCT-2715

---
### How to test

Everything should be functioning as expected

### Important updates

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Comment by gmrabian** [comment]
❗

File: services/ui-src/src/utils/testing/setupJest.tsx [hunk 1]
```diff
@@ -42,18 +42,59 @@ export const mockLDFlags = {
...
+// USER STATES
+
+export const mockNoUserStore: MfpUserState = {
+  user: null,
   showLocalLogins: true,
-  logout: async () => {},
+  setUser: () => {},
+  setShowLocalLogins: () => {},
   loginWithIDM: () => {},
-  updateTimeout: async () => {},
```
Comment (karla-vm) [inline services/ui-src/src/utils/testing/setupJest.tsx:52]: @braxex @BearHanded my biggest concern / question is about removing these from Context going forward and how they'll be used in the store.

File: services/ui-src/src/utils/state/useUser.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { create } from "zustand";
+import { devtools, persist } from "zustand/middleware";
+// types
+import { MfpUserState, MFPUser } from "types";
+
+export const useUser = create(
```
Comment (BearHanded) [inline services/ui-src/src/utils/state/useUser.ts:6]: General zustand question: Should state initialization still be a use____ style hook? Or is this something that only needs to be initialized one time?

**Review by britt-mo — APPROVED** [review]

**Review by ntsummers1 — APPROVED** [review]
LGTM!

**Review by BearHanded — APPROVED** [review]

**Pull request merged**